src/citationnet/static/force-citationNet.js

```diff
@@ -0,0 +1,419 @@
+        this.inputNode = data.nodes.filter(o => o.attributes.is_input_DOI == true)[0];
+        var inputNode = this.inputNode;
+
+        for (let i = 0; i < data.nodes.length; i++) {
+            id_map[data.nodes[i].id] = i;
+            data.nodes[i].outgoingLinks = [];
+            data.nodes[i].outgoingLinkTo = [];
+            data.nodes[i].incomingLinks = [];
+            data.nodes[i].incomingLinkFrom = [];
+
+            // delete unused attributes
+            delete data.nodes[i].color;
+            delete data.nodes[i].size;
+            delete data.nodes[i].x;
+            delete data.nodes[i].y;
+
+            // fix z-coordinate of nodes depending on publication year
+            // 20 units between input node and years before/after
+            // 5 units spacing between years
+            if (data.nodes[i].attributes.nodeyear >= inputNode.attributes.nodeyear) {
+                data.nodes[i].fz = 5 * (data.nodes[i].attributes.nodeyear - inputNode.attributes.nodeyear + 20);
+            } else {
+                data.nodes[i].fz = 5 * (data.nodes[i].attributes.nodeyear - inputNode.attributes.nodeyear - 20);
+            }
+        }
+
+        // fix position of input node, color red
+        inputNode.fx = 0.0;
+        inputNode.fy = 0.0;
+        inputNode.fz = 0.0;
+        inputNode.color = 'red';
+
+        // cross-link node objects
+        data.edges.forEach(edge => {
+            var a = data.nodes[id_map[edge.source]];
+            var b = data.nodes[id_map[edge.target]];
+            if (typeof a != "undefined" && typeof b != "undefined") {
+              !a.outgoingLinks && (a.outgoingLinks = []);
+              a.outgoingLinks.push(edge);
+
+              !a.outgoingLinkTo && (a.outgoingLinkTo = []);
+              a.outgoingLinkTo.push(b);
+
+              !b.incomingLinks && (b.incomingLinks = []);
+              b.incomingLinks.push(edge);
+
+              !b.incomingLinkFrom && (b.incomingLinkFrom = []);
+              b.incomingLinkFrom.push(a);
+
+              delete edge.color;
+              delete edge.size;
+            }
+        }
+      );
+    }
+
+    /**
+    * Move camera to default view point. Triggered by UI.
+    * @param {String} viewPoint - either "top" or "side"
+    * @returns {ReturnValueDataTypeHere} Brief description of the returning value here.
+    */
+    view(viewPoint) {
+        if (viewPoint == 'top') {
+            // indicate view point using bold font
+            document.getElementById("btnTopView").style.fontWeight = "bold";
+            document.getElementById("btnSideView").style.fontWeight = "normal";
+            // set camera position, zoom and viewing direction (up-vector)
+            this.graph.cameraPosition({ x: 0, y: 0, z: 500 }, { x: 0, y: 0, z: 0 }, 500);
+            this.graph.camera().up.set(0.0, 1.0, 0.0);
+            this.graph.camera().zoom = 2.0;
+            this.graph.camera().updateProjectionMatrix();
+        } else if (viewPoint == 'side') {
```

Comment:
`==` is a "loose equality" and can cause issues in JavaScript. Maybe not in this specific example, but it's generally safer to use `===` (strict equality) that checks both the value and the type of the terms

Reply:
Example:

```
> "" == false
true
> "" === false
false
```